testing/sliding-sync-integration-test/src/lib.rs

```diff
@@ -0,0 +1,797 @@
+use matrix_sdk::{
+    ruma::api::client::room::create_room::v3::Request as CreateRoomRequest, Client, RoomListEntry,
+    SlidingSyncBuilder,
+};
+use matrix_sdk_integration_testing::helpers::get_client_for_user;
+
+#[allow(dead_code)]
+async fn setup(name: String) -> anyhow::Result<(Client, SlidingSyncBuilder)> {
+    let sliding_sync_proxy_url =
+        option_env!("SLIDING_SYNC_PROXY_URL").unwrap_or("http://localhost:8338").to_owned();
+    let client = get_client_for_user(name, false).await?;
+    let sliding_sync_builder = client
+        .sliding_sync()
+        .await
+        .homeserver(sliding_sync_proxy_url.parse()?)
+        .with_common_extensions();
+    Ok((client, sliding_sync_builder))
+}
+
+#[allow(dead_code)]
+async fn random_setup_with_rooms(
+    number_of_rooms: usize,
+) -> anyhow::Result<(Client, SlidingSyncBuilder)> {
+    let namespace = uuid::Uuid::new_v4().to_string();
+    let (client, sliding_sync_builder) = setup(namespace.clone()).await?;
+
+    for room_num in 0..number_of_rooms {
+        let mut request = CreateRoomRequest::new();
+        request.name = Some(format!("{namespace}-{room_num}"));
+        let _event_id = client.create_room(request).await?;
+    }
+
+    Ok((client, sliding_sync_builder))
+}
+
+#[derive(PartialEq, Eq, Debug)]
+enum RoomListEntryEasy {
+    Empty,
+    Invalid,
+    Filled,
+}
+
+impl From<&RoomListEntry> for RoomListEntryEasy {
+    fn from(value: &RoomListEntry) -> Self {
+        match value {
+            RoomListEntry::Empty => RoomListEntryEasy::Empty,
+            RoomListEntry::Invalidated(_) => RoomListEntryEasy::Invalid,
+            RoomListEntry::Filled(_) => RoomListEntryEasy::Filled,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use futures::{pin_mut, stream::StreamExt};
+    use matrix_sdk::{
+        ruma::events::room::message::RoomMessageEventContent, SlidingSyncMode,
+        SlidingSyncViewBuilder,
+    };
+
+    use super::*;
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+    async fn it_works_smoke_test() -> anyhow::Result<()> {
+        let (_client, sync_proxy_builder) = setup("odo".to_owned()).await?;
+        let sync_proxy = sync_proxy_builder.add_fullsync_view().build().await?;
+        let stream = sync_proxy.stream().await?;
+        pin_mut!(stream);
+        let Some(room_summary ) = stream.next().await else {
+            anyhow::bail!("No room summary found, loop ended unsuccessfully");
+        };
```

Comment:
If you

```rust
use anyhow::Context;
```

you can do

```suggestion
        room_summary = stream.next().await.context("No room summary found, loop ended unsuccessfully")?;
```

hree and in most / all the places below that use `bail!`.

Reply:
So now you imported `bail!` instead but kept it? I'm a little confused 😅

Reply:
sorry. I misread. I changed most occurrences - but the `Some(RoomListEntry::Filled(room_id))` stuck because the compiler didn't accept it otherwise.

Reply:
Okay, can you revert the new `panic`s to `bail!` though? Surely we don't want to panic when we could return an error?

Reply:
Oh, this is test code so doesn't really matter.